src/libsyntax/codemap.rs

```diff
@@ -592,16 +592,21 @@ impl CodeMap {
         CodeMap {
             files: RefCell::new(Vec::new()),
             expansions: RefCell::new(Vec::new()),
-            file_loader: file_loader
+            file_loader: RefCell::new(file_loader)
         }
     }
 
+    pub fn set_file_loader(&self, file_loader: Box<FileLoader>) {
+        let mut loader = self.file_loader.borrow_mut();
+        *loader = file_loader;
+    }
+
```

Comment:
Don't you need a custom diagnostics handler? In my [custom `rustc` driver](https://github.com/eddyb/r3/blob/a81d690a4831094e5e276ec58e8a9965393d2363/src/ide/rustc.rs#L163) I had to create everything myself, so it didn't seem unreasonable to have to pass the file loader when creating the `CodeMap`.